packages/agent-base/src/index.ts

```diff
@@ -21,21 +21,37 @@ export type AgentConnectOpts = HttpConnectOpts | HttpsConnectOpts;
 const INTERNAL = Symbol('AgentBaseInternalState');
 
 interface InternalState {
+	tlsUpgradeOpts?: TlsUpgradeOpts;
 	defaultPort?: number;
 	protocol?: string;
 	currentSocket?: Duplex;
 }
 
+export interface TlsUpgradeOpts {
+	cert?: string;
+	key?: string;
+	ca?: string;
+}
```

Comment:
The are a whole bunch of other config options that can be used. How about we use this instead?

```suggestion
export type TlsUpgradeOpts = Omit<tls.ConnectionOptions, 'path' | 'port' | 'host' | 'socket' | 'servername'>;
```